enos/enos.py

```diff
@@ -182,6 +182,11 @@ def up(provider=None, env=None, **kwargs):
 def install_os(env=None, **kwargs):
     logging.debug('phase[os]: args=%s' % kwargs)
 
+    if 'provider_net' not in env:
```

Comment:
Maybe we could have a util function that check it. This way we could unit test this part easily.
We want to check two things
* if a key exist in a dict
* or if the value associated is not None in some cases

What about having `check_env(env, present=[], notNone=[])` where `env` is the environnement to check, `present` are the keys that needs to be present and `notNone` are the key whose values can't be None. This function throws an exception where something wrong is found.